grocery-app/src/pages/SignIn/SignUp.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from "react";
+import "./signup.css";
+import sidenav from "./sideimage3.jpg";
+import { Link } from "react-router-dom";
+import { signupServer } from "../../services/auth_services";
+import { useNavigate } from "react-router-dom";
+function SignUp() {
+  const [userObj, setUserObj] = useState({
+    username: "",
+    first_name: "",
+    last_name: "",
+    email: "",
+    password: "",
+  });
+  const [isValid, setIsValid] = useState(false);
+  let navigate = useNavigate();
+
+  const validateEmail = (event) => {
+    setUserObj((userObj) => ({ ...userObj, email: event.target.value }));
+    const emailRegex = /^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$/;
+    if (emailRegex.test(userObj.email)) {
```

Comment:
for simplification you can write this condition as - setIsValid(emailRegex.test(userObj.email)) this will reduce 4 lines of code

Reply:
done..